captain/parser/bool_parser/expressions/exceptions.py

```diff
@@ -44,3 +44,9 @@ class InvalidIdentifier(Exception):
     def __init__(self, message=""):
         self.message = "Invalid identifier: " + message
         super().__init__(self.message)
+
+
+class EarlyIdentifier(Exception):
+    def __init__(self, message=""):
+        self.message = f"Cannot access identifier {message} before evaluation"
```

Comment:
The error is good, but a little bit too "comp sci" for a user. Maybe something like:
```
f"Unable to access test result for {message} as it's executed afterward"
```